### PR TITLE
Security Measures

### PR DESCRIPTION
**Pull request by matthewrsj**
The following security measures have been implemented:
* Validate email address from the form
* Check that the name field has allowed characters
* Use a token to verify the app is talking to the correct form
* Create a hidden field in the form - if this field is filled out, a robot likely submitted the form

The tests for these features all pass. The form does not send an email unless all these are validated.

### CODE REVIEW

**Comment by matthewrsj** [comment]
@Kennric can you please review

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+language: python
+
+python:
+  - "2.6"
+  - "2.7"
+  - "3.2"
+  - "3.3"
+  - "3.4"
```
Comment (tschuy) [inline .travis.yml]: Do we really want to support five versions of python?

File: templates/index.html [hunk 2]
```diff
@@ -6,8 +6,10 @@
         <form enctype="multipart/form-data" action="" method="post" accept-charset="UTF-8">
             <label for="edit-submitted-name">Name</label>
             <input type="text" name="name" value="" size="60" maxlength="128" />
-            <label for="edit-submitted-age">Age</label>
-            <input type="text" name="age" value="" size="60" maxlength="128" />
+            <label for="edit-submitted-email">Email</label>
+            <input type="text" name="email" value="" size="60" maxlength="128" />
+            <input type="hidden" name="hidden" value="" size="60" maxlength="128" />
+            <input type="hidden" name="tokn" value="1234567890-=" size="60" maxlength="128" />
```
Comment (tschuy) [inline templates/index.html]: The way that TOKN works will be that it's statically rendered on the front end and then checked against the backend, correct?

File: .travis.yml (hunk 1 above)
Comment (matthewrsj) [inline .travis.yml]: Probably not. Should I go with most recent?

File: templates/index.html (hunk 2 above)
Comment (matthewrsj) [inline templates/index.html]: correct

File: request_handler.py [hunk 3]
```diff
@@ -78,6 +96,33 @@ def create_msg(request):
...
+                                 verify=True)
+    if valid_email:
+        return valid_email
+    return False
+
+
+def validate_name(request):
+    name = request.form['name']
+    name_with_no_spaces = name.replace(" ", "")
+    if name_with_no_spaces.isalpha():
```
Comment (tschuy) [inline request_handler.py]: http://www.kalzumeus.com/2010/06/17/falsehoods-programmers-believe-about-names/

Names aren't alpha. Zoé, Clément, and 昭夫 wouldn't appreciate this decision. :negative_squared_cross_mark:

Comment (tschuy) [inline request_handler.py]: What do we want to validate a name for, anyway?

Comment (matthewrsj) [inline request_handler.py]: Yeah. @Kennric what characters did you want me to exclude from the name field? I was wondering about this too.

Comment (tschuy) [inline request_handler.py]: I feel like
```
if name.strip() return True
```

should be good -- just check there's text there. It's not like we're going to be [texting this to people](http://www.telegraph.co.uk/technology/apple/11633176/This-text-will-make-your-friends-iPhone-crash-if-you-send-it-to-them-heres-how-to-protect-your-own-phone.html)...

Comment (matthewrsj) [inline request_handler.py]: We wanted to add another layer of security, we were looking for a way to check if it is obviously not a human filling out the form.

File: .travis.yml (hunk 1 above)
Comment (tschuy) [inline .travis.yml]: I would support 2.7 and 3.4, or just one of them. Most of our stuff is 2.7-only or 2.6/2.7 but Python 3 is the language of the future.

**Comment by matthewrsj** [comment]
@tschuy @Kennric made those changes

**Comment by Kennric** [comment]
+1

**Comment by tschuy** [comment]
:+1:

**Pull request merged**